main.nf

```diff
@@ -0,0 +1,932 @@
+ * 
+*/
+process variantAnnotatesnpEff {
+    tag "$name"
+    publishDir "${params.outdir}/annotatedVariants/", mode: 'copy'
+
+    input:
+    set file(phased_vcf), file(phased_vcf_ind) from combined_variants_snpEff
+
+    output:
+    file "*.{snpeff}" into combined_variants_gatk_snpeff
+    file '.command.log' into snpeff_stdout
+    file 'SnpEffStats.csv' into snpeff_results
+
+    script:
+    """
+        snpEff \\
+        -c /usr/local/lib/snpEff/snpEff.config \\
+        -i vcf \\
+        -csvStats SnpEffStats.csv \\
+        -o gatk \\
+        -o vcf \\
+        -filterInterval $params.target_bed GRCh37.75 $phased_vcf \\
+            > ${name}_combined_phased_variants.snpeff 
+        
+        # Print version number to standard out
+        echo "GATK version "\$(snpEff -version 2>&1)
+    """
+}
+
+
+/*
+ * Step 16 - Annotate Variants with GATK
+ * 
+*/
+
+process variantAnnotateGATK{     
+    tag "$name"
+    publishDir "${params.outdir}/annotatedVariants/", mode: 'copy'
+
+    input:
+    set file(phased_vcf), file(phased_vcf_ind) from combined_variants_gatk
+    file(phased_vcf_snpeff) from combined_variants_gatk_snpeff
+
+    output:
+    file "*.{vcd,idx}"
+
+    script:
+    """
+    gatk -T VariantAnnotator \\
+        -R $params.gfasta \\
+        -A SnpEff \\
+        --variant $phased_vcf \\
+        --snpEffFile ${name}_combined_phased_variants.snpeff \\
+        --out ${name}_combined_phased_annotated_variants.vcf
+    """
+}
+
+
+
+/*
+ * Step 17 - Perform variant evaluation with GATK's VariantEval tool
+*/
+
+process variantEvaluate {
+    tag "$name"
+    publishDir "${params.outdir}/${name}/variants", mode: 'copy'
+
+    input:
+    set file("${name}_combined_variants.vcf"), file("${name}_combined_variants.vcf.idx") from combined_variants_evaluate
+
+    output:
+    file "${name}_combined_phased_variants.eval"
+    file "${name}_combined_phased_variants.eval" into gatk_variant_eval_results
+
+    script:
+    """
+    gatk -T VariantEval \\
+        -R $params.gfasta \\
+        --eval $phased_vcf \\
+        --dbsnp $params.dbsnp \\
+        -o ${name}_combined_phased_variants.eval \\
+        -L $params.target \\
+        --doNotUseAllStandardModules \\
+        --evalModule TiTvVariantEvaluator \\
+        --evalModule CountVariants \\
+        --evalModule CompOverlap \\
+        --evalModule ValidationReport \\
+        --stratificationModule Filter \\
+        -l INFO
+    """
+}
+
+
+/*
+ * Step 18 - Generate Software Versions Map
+ * 
+*/
+software_versions = [
+  'FastQC': null, 'Trim Galore!': null, 'BWA': null, 'Picard MarkDuplicates': null, 'GATK': null,
+  'SNPEff': null, 'QualiMap': null, 'Nextflow': "v$workflow.nextflow.version"
+]
+
+/*
+* Step 19 - Generate a YAML file for software versions in the pipeline
+* This is then parsed by MultiQC and the report feature to produce a final report with the software Versions in the pipeline.
+*/ 
+
+process get_software_versions {
```

Comment:
For reference, I don't advise collecting software versions from stdout like this any more. It's messy and kind of a pain - having a single process to generate them all and a python script to generate the yaml is much cleaner (as with the [NGI-MethylSeq pipeline](https://github.com/SciLifeLab/NGI-MethylSeq/blob/196c42fb4d0401f5b33e9d8e828cf74396cc26e5/bismark.nf#L498-L524).

Not sure how that approach would work with multiple containers, mind.

Reply:
Okay. I added a issue for that purpose, but might need some time to rethink how to achieve this properly.